src/ir_Coolix.cpp

```diff
@@ -215,125 +275,144 @@ void IRCoolixAC::off(void) {
-  updateSavedState();
-  remote_state = kCoolixLed;
+void IRCoolixAC::setSleep(const bool sendflag) {
+  sleepFlag = !sleepFlag;
+  if(sendflag)
+    send(kCoolixSleep,kCoolixBits);
 }
 
-bool IRCoolixAC::getClean() { return remote_state == kCoolixClean; }
+bool IRCoolixAC::getTurbo() { return turboFlag; }
 
-void IRCoolixAC::setClean() {
-  updateSavedState();
-  remote_state = kCoolixClean;
+void IRCoolixAC::setTurbo(const bool sendflag) {
+  turboFlag = !turboFlag;
+  if(sendflag)
+    send(kCoolixTurbo,kCoolixBits);
+}
+
+bool IRCoolixAC::getLed() { return ledFlag; }
+
+void IRCoolixAC::setLed(const bool sendflag) {
+  ledFlag = !ledFlag;
+  if(sendflag)
+    send(kCoolixLed,kCoolixBits);
+}
+
+bool IRCoolixAC::getClean() { return cleanFlag; }
+
+void IRCoolixAC::setClean(const bool sendflag) {
+  cleanFlag = !cleanFlag;
+  if(sendflag)
+    send(kCoolixClean,kCoolixBits);
 }
 
 bool IRCoolixAC::getZoneFollow() {
-  return getNormalState() & kCoolixZoneFollowMask;
+  return zoneFollowFlag;
 }
 
 // Internal use only.
 void IRCoolixAC::setZoneFollow(bool state) {
-  recoverSavedState();
   if (state) {
-    remote_state |= kCoolixZoneFollowMask;
+    remote |= kCoolixZoneFollowMask;
   } else {
-    remote_state &= ~kCoolixZoneFollowMask;
+    remote &= ~kCoolixZoneFollowMask;
   }
+  // update flag
+  zoneFollowFlag = state;
 }
 
 void IRCoolixAC::clearSensorTemp() {
-  recoverSavedState();
   setZoneFollow(false);
   setSensorTempRaw(kCoolixSensorTempIgnoreCode);
 }
 
+uint8_t IRCoolixAC::getMode() {
+  return acMode;
+}
+
 void IRCoolixAC::setMode(const uint8_t mode) {
   uint32_t actualmode = mode;
+  acMode = mode;
   switch (actualmode) {
     case kCoolixAuto:
     case kCoolixDry:
-      if (this->getFan() == kCoolixFanAuto)
-        //  No kCoolixFanAuto in Dry/Auto mode.
         this->setFan(kCoolixFanAuto0, false);
       break;
     case kCoolixCool:
     case kCoolixHeat:
     case kCoolixFan:
-      if (this->getFan() == kCoolixFanAuto0)
-        // kCoolixFanAuto0 only in Dry/Auto mode.
         this->setFan(kCoolixFanAuto, false);
       break;
     default:  // Anything else, go with Auto mode.
-      this->setMode(kCoolixAuto);
+        this->setMode(kCoolixAuto);
       return;
   }
   // Fan mode is a special case of Dry.
-  if (mode == kCoolixFan) actualmode = kCoolixDry;
-  recoverSavedState();
-  remote_state = (remote_state & ~kCoolixModeMask) | (actualmode << 2);
-  // Force the temp into a known-good state.
-  setTemp(getTemp());
-  if (mode == kCoolixFan) setTempRaw(kCoolixFanTempCode);
-}
-
-uint8_t IRCoolixAC::getMode() {
-  uint8_t mode = (getNormalState() & kCoolixModeMask) >> 2;
-  if (mode == kCoolixDry)
-    if (getTempRaw() == kCoolixFanTempCode) return kCoolixFan;
-  return mode;
+  if (mode == kCoolixFan){
+    actualmode = kCoolixDry;
+    setTempRaw(kCoolixFanTempCode);
+  }
+  // set
+  remote &= ~kCoolixModeMask;
+  remote |= (actualmode << 2);
 }
 
 uint8_t IRCoolixAC::getFan() {
```

Comment:
Add a flag for if you want to get this from the raw `remote` code, or from the variable.
e.g. `uint8_t IRCoolixAC::getFan(const bool raw = true)`

This keeps the bit logic/math in the `getFan()` method, rather in the `setRaw()` method.

ditto for `getMode()`